File: week3/Eren-Taskiran/141-linked-list-cycle.c
bool hasCycle(struct ListNode *head) {
     if (head == NULL || head->next == NULL) {
        return false;
    }
    
    struct ListNode *slow = head,*fast = head->next;
    
    while (slow != fast) {
        if (fast == NULL || fast->next == NULL) {
            return false;
        }
        slow = slow->next;
        fast = fast->next->next;
    }
    
    return true;
}

File: week3/Can-Yavuz/141-linked-list-cycle.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public bool HasCycle(ListNode head) {
         if (head == null || head.next == null) {
            return false; 
        }

        ListNode slow = head, fast = head.next;
        
         while (slow != fast) {
            if (fast == null || fast.next == null) {
                return false; 
            }
            slow = slow.next;
            fast = fast.next.next;
        }

        return true; 
    }
}